src/include/cursor.i

```diff
@@ -178,7 +178,7 @@ __cursor_leave(WT_SESSION_IMPL *session)
      * active cursors, and we can release any snapshot we're holding for read committed isolation.
      */
     WT_ASSERT(session, session->ncursors > 0);
-    if (--session->ncursors == 0)
+    if (--session->ncursors == 0 && !F_ISSET(session, WT_CBT_NO_TXN))
```

Comment:
The `WT_CBT_NO_TXN` flag only applies to `WT_CURSOR_BTREE` pointers, it shouldn't apply to session pointers.  Is there any reason this is check needs to change?

Reply:
@ddanderson Thanks for explaining to me the background of cbt and session pointers. I have removed the checking from `__cursor_leave` as suggested.

Reply:
I think this check was necessary. We want WT_CBT_NO_TXN to result in cursor tracking happening, but snapshot allocation/free being skipped.
Those two things are correlated here - I don't think __wt_txn_read_last is setup to accept a transaction that doesn't have a snapshot allocated, and skipping the txn_cursor_op call when the WT_CBT_NO_TXN flag means we won't have a snapshot here.

Reply:
@agorrod, given the other changes in this PR, I don't think `__cursor_leave` is ever called with `WT_CBT_NO_TRACKING` on (used to be `WT_CBT_NO_TXN`).  Well, I do see `__cursor_leave` being called in LSM.  Perhaps a cleaner approach is change `__cursor_enter` and `__cursor_leave` to have a cursor argument, and check the new tracking flag within those functions.